src/Nri/Ui/Button/V9.elm

```diff
@@ -0,0 +1,887 @@
+
+-}
+linkSpa : Attribute msg
+linkSpa buttonOrLink =
+    buttonOrLink
+
+
+{-| Wrap some text so it looks like a button, but actually is wrapped in an anchor to
+some url, and it's an HTTP request (Rails includes JS to make this use the given HTTP method)
+-}
+linkWithMethod : String -> Attribute msg
+linkWithMethod method (ButtonOrLink config) =
+    ButtonOrLink { config | linkType = WithMethod method }
+
+
+{-| Wrap some text so it looks like a button, but actually is wrapped in an anchor to some url.
+This should only take in messages that result in a Msg that triggers Analytics.trackAndRedirect. For buttons that trigger other effects on the page, please use Nri.Button.button instead
+-}
+linkWithTracking : Attribute msg
+linkWithTracking (ButtonOrLink config) =
+    ButtonOrLink { config | linkType = WithTracking }
+
+
+{-| Wrap some text so it looks like a button, but actually is wrapped in an anchor to
+some url and have it open to an external site
+-}
+linkExternal : Attribute msg
+linkExternal (ButtonOrLink config) =
+    ButtonOrLink { config | linkType = External }
+
+
+{-| Wrap some text so it looks like a button, but actually is wrapped in an anchor to some url and have it open to an external site
+
+This should only take in messages that result in tracking events. For buttons that trigger other effects on the page, please use Nri.Ui.Button.V2.button instead
+
+-}
+linkExternalWithTracking : Attribute msg
+linkExternalWithTracking (ButtonOrLink config) =
+    ButtonOrLink { config | linkType = ExternalWithTracking }
+
+
+
+-- BUTTON SIZING
+
+
+{-| -}
+small : Attribute msg
+small (ButtonOrLink config) =
+    ButtonOrLink { config | size = Small }
+
+
+{-| -}
+medium : Attribute msg
+medium (ButtonOrLink config) =
+    ButtonOrLink { config | size = Medium }
+
+
+{-| -}
+large : Attribute msg
+large (ButtonOrLink config) =
+    ButtonOrLink { config | size = Large }
+
+
+
+-- BUTTON WIDTH
+
+
+type ButtonWidth
+    = WidthExact Int
+    | WidthUnbounded
+    | WidthFillContainer
+
+
+{-| Sizes for buttons and links that have button classes
+-}
+type ButtonSize
+    = Small
+    | Medium
+    | Large
+
+
+{-| Define a size in `px` for the button's total width.
+-}
+exactWidth : Int -> Attribute msg
+exactWidth inPx (ButtonOrLink config) =
+    ButtonOrLink { config | width = WidthExact inPx }
+
+
+{-| Leave the maxiumum width unbounded (there is a minimum width).
+-}
+unboundedWidth : Attribute msg
+unboundedWidth (ButtonOrLink config) =
+    ButtonOrLink { config | width = WidthUnbounded }
+
+
+{-| -}
+fillContainerWidth : Attribute msg
+fillContainerWidth (ButtonOrLink config) =
+    ButtonOrLink { config | width = WidthFillContainer }
+
+
+
+-- COLOR SCHEMES
+
+
+type ButtonStyle
+    = Primary
+    | Secondary
+    | Danger
+    | Premium
+      -- The following styles cannot be set directly.
+      -- Use a helper like `unfulfilled` to change the state instead.
```

Comment:
🎨 I think it would be clearer to have two distinct type, one for the 4 "styles", and then a separate one with all 8 "color schemes" if a type for the scheme is really even necessary

Reply:
(or maybe the view code could just destructure the Style and the State at the same time and then having the extra "schemes" as custom type variants wouldn't even be necessary?)